catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/dialog/NewProjectDialogFragment.java

```diff
@@ -40,79 +40,96 @@
+
 	@Override
 	public Dialog onCreateDialog(Bundle savedInstanceState) {
 		View view = View.inflate(getActivity(), R.layout.dialog_new_project, null);
 
 		final RadioGroup radioGroup = view.findViewById(R.id.radio_group);
+		final Switch exampleProjectSwitch = view.findViewById(R.id.example_project_switch);
 
-		if (SettingsFragment.isDroneSharedPreferenceEnabled(getContext())) {
-			view.findViewById(R.id.project_default_drone_radio_button).setVisibility(View.VISIBLE);
-		}
-
-		if (JumpingSumoServiceWrapper.isJumpingSumoSharedPreferenceEnabled()) {
-			view.findViewById(R.id.project_default_jumping_sumo_radio_button).setVisibility(View.VISIBLE);
+		if (SettingsFragment.isCastSharedPreferenceEnabled(getActivity())) {
+			view.findViewById(R.id.cast).setVisibility(View.VISIBLE);
 		}
 
 		TextInputDialog.Builder builder = new TextInputDialog.Builder(getContext())
 				.setHint(getString(R.string.project_name_label))
+				.setText(getUniqueDefaultProjectName())
 				.setTextWatcher(new NewProjectNameTextWatcher<>())
 				.setPositiveButton(getString(R.string.ok), (TextInputDialog.OnClickListener) (dialog, textInput) -> {
+					if (exampleProjectSwitch.isChecked()) {
+						exampleProject = true;
+					} else {
+						exampleProject = false;
+					}
+
 					switch (radioGroup.getCheckedRadioButtonId()) {
-						case R.id.project_empty_radio_button:
-							showOrientationDialog(textInput, true);
-							break;
-						case R.id.project_default_radio_button:
-							showOrientationDialog(textInput, false);
+						case R.id.portrait:
+							landscape = false;
 							break;
-						case R.id.project_default_drone_radio_button:
-							createARDroneProject(textInput);
+						case R.id.landscape_mode:
+							landscape = true;
 							break;
-						case R.id.project_default_jumping_sumo_radio_button:
-							createJumpingSumoProject(textInput);
+						case R.id.cast:
+							castProject = true;
 							break;
 						default:
 							throw new IllegalStateException(TAG + ": No radio button id match, check layout?");
 					}
+
+					createProject(textInput, landscape, exampleProject, castProject);
 				});
 
 		return builder
-				.setTitle(R.string.new_project_dialog_title)
 				.setView(view)
 				.setNegativeButton(R.string.cancel, null)
 				.create();
 	}
 
-	void showOrientationDialog(String projectName, boolean createEmptyProject) {
-		OrientationDialogFragment
-				.newInstance(projectName, createEmptyProject)
-				.show(getFragmentManager(), OrientationDialogFragment.TAG);
-	}
+	public String getUniqueDefaultProjectName() {
 
-	void createARDroneProject(String name) {
-		try {
-			ProjectManager.getInstance()
-					.createNewExampleProject(name, getContext(), PROJECT_CREATOR_DRONE, false);
-			Intent intent = new Intent(getActivity(), ProjectActivity.class);
-			startActivity(intent);
-		} catch (IOException e) {
-			ToastUtil.showError(getActivity(), R.string.error_new_project);
+		for (int i = 1; i < Integer.MAX_VALUE; i++) {
```

Comment:
Please have a look at the UniqueNameProvider class. Getting a unique name with ascending values is already implemented there. Use something like return new UniqueNameProvider(default_project_name, listOfProjectNames).

Reply:
The UniqueNameProvider class does not match the required pattern for the name, it would return "My project (1)" instead of "My project 1".